src/services/aws/aws.ts

```diff
@@ -139,30 +126,40 @@ export const {
   },
 } = aws;
 
-type EndowCardFields = keyof (Omit<EndowmentCard, "hq" | "categories"> &
-  /** replace with cloudsearch specific field format */
-  Pick<EndowmentProfileUpdate, "hq_country" | "categories_sdgs">);
-
-//object format first to avoid duplicates
-const endowCardObj: {
-  [key in EndowCardFields]: any; //we care only for keys
-} = {
-  hq_country: "",
-  active_in_countries: "",
-  categories_sdgs: "",
-  id: "",
-  image: "",
-  kyc_donors_only: "",
-  name: "",
-  tagline: "",
-  endow_type: "",
+export const useEndowmentsQuery = <T extends Partial<Endowment>>(
```

Comment:
> also CharityHeader for this matter would not get data from v3/endowments but from some profile/{endowId} so this is a farfetched example

`CharityHeader` already exists and fetches this data, we just didn't create a separate endpoint with a specific set of fields for it (exactly my point).

> The Card's job is to display data from the endpoint, it shoudn't concern itself on projections

Point of this PR was to exactly invert this relationship and make it easier to reuse the same endpoint while fetching just the (amount of) data needed for a specific component, in this case `Card` and `EndowmentCombobox`. Otherwise we end up creating those funny-looking endpoints like `useEndowmentIdNameQuery`.

> problem is you do this refactors without first creating a ticket to discuss if even there's a need for this given our priorities

You mean like you did with [Component styles abstractions with css layers](https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1678)? That was a very big change to our codebase that had **no priority, no discussion prior to it and no ticket attached**. It was merged as a bundle of many changes, some of which we didn't all agree (and I still don't agree) were necessary (e.g. creating `btn-gift, btn-donate, btn-reg` etc. in the root `index.css`, that would have been better off as HOC).
To do that and then lecture others is hypocritical. 

This PR introduces a new level of abstraction to allow reusing the same endpoint while maintaining performance and maintain the amount of data transferred over the network. Having an extra 50 lines of code to remove `useEndowmentCardsQuery` (doesn't return "endowment cards" but returns just "endowments") and `useEndowmentIdNameQuery` (ridiculous ad-hoc name).

But point taken.

@ap-justin

Reply:
> You mean like you did with https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1678?
seriously? bringing back old PR just to make this one pass ? 

I have learned from that, and never did again. and to tell you the difference
1. I am doing all that extra work on my free time - making sure I'm still focusing on tasks priotized by Andrey
2. I am actually solving a burden that's prevalent in the codebase (giftcard, registration, donate page), unlike this PR, fixing something that isn't broken / improving something that the need is just imagined

Again, there's no ticket for and even this changes are made/shown - It's not evident why we need to merge this code. Can you atl east relate this to a tasks ? not just a ticket you created on your own ?